sunpy/util/decorators.py

```diff
@@ -325,3 +325,52 @@ def inner_f(*args, **kwargs):
             return f(**kwargs)
         return inner_f
     return deprecate_positional_args
+
+
+_NOT_FOUND = object()
+
+
+def cached_property_based_on(meth):
+    """
+    A decorator to cache the value of a property based on the output of a
+    different callable.
+
+    This decorator caches the value of ``meth(instance)`` and ``prop(instance)``.
+    When the decorated property is accessed, ``meth(instance)`` is recomputed.
+    If this returns the same as its cached value, the cached value of ``prop``
+    is returned. Otherwise both ``meth`` and ``prop`` are recomputed, cached,
+    and the new value of ``prop`` is returned.
+
+    Parameters
+    ----------
+    meth : callable
+        A function that takes exactly one argument (a class instance) and
+        returns an arbitrary output. Changes in the output are used to decide
+        whether or not to recompute the property.
+
+    Notes
+    -----
+    The cached value of ``meth(instance)`` is stored under the key ``meth.__name__``.
+    """
+    def outer(prop):
```

Comment:
Just remembered I forgot to copy the docstring across, leaving as a not for myself.

Reply:
doesn't `wraps` do that?

Reply:
Yep, this was a comment from before I added wraps